tests/classification/test_jaccard.py

```diff
@@ -104,60 +104,61 @@ def _sk_jaccard_multidim_multiclass(preds, target, average=None):
+        # average = "macro" if reduction == "elementwise_mean" else None  # convert tags
         self.run_class_metric_test(
             ddp=ddp,
             preds=preds,
             target=target,
             metric_class=JaccardIndex,
             sk_metric=partial(sk_metric, average=average),
             dist_sync_on_step=dist_sync_on_step,
-            metric_args={"num_classes": num_classes, "threshold": THRESHOLD, "reduction": reduction},
+            metric_args={"num_classes": num_classes, "threshold": THRESHOLD, "average": average},
         )
 
-    def test_jaccard_functional(self, reduction, preds, target, sk_metric, num_classes):
-        average = "macro" if reduction == "elementwise_mean" else None  # convert tags
+    def test_jaccard_functional(self, average, preds, target, sk_metric, num_classes):
+        # average = "macro" if reduction == "elementwise_mean" else None  # convert tags
         self.run_functional_metric_test(
             preds,
             target,
             metric_functional=jaccard_index,
             sk_metric=partial(sk_metric, average=average),
-            metric_args={"num_classes": num_classes, "threshold": THRESHOLD, "reduction": reduction},
+            metric_args={"num_classes": num_classes, "threshold": THRESHOLD, "average": average},
         )
 
-    def test_jaccard_differentiability(self, reduction, preds, target, sk_metric, num_classes):
+    def test_jaccard_differentiability(self, average, preds, target, sk_metric, num_classes):
         self.run_differentiability_test(
             preds=preds,
             target=target,
             metric_module=JaccardIndex,
             metric_functional=jaccard_index,
-            metric_args={"num_classes": num_classes, "threshold": THRESHOLD, "reduction": reduction},
+            metric_args={"num_classes": num_classes, "threshold": THRESHOLD, "average": average},
         )
 
 
 @pytest.mark.parametrize(
-    ["half_ones", "reduction", "ignore_index", "expected"],
+    ["half_ones", "average", "ignore_index", "expected"],
     [
         (False, "none", None, Tensor([1, 1, 1])),
-        (False, "elementwise_mean", None, Tensor([1])),
+        (False, "macro", None, Tensor([1])),
         (False, "none", 0, Tensor([1, 1])),
         (True, "none", None, Tensor([0.5, 0.5, 0.5])),
-        (True, "elementwise_mean", None, Tensor([0.5])),
+        (True, "macro", None, Tensor([0.5])),
         (True, "none", 0, Tensor([2 / 3, 1 / 2])),
     ],
 )
-def test_jaccard(half_ones, reduction, ignore_index, expected):
+def test_jaccard(half_ones, average, ignore_index, expected):
     preds = (torch.arange(120) % 3).view(-1, 1)
     target = (torch.arange(120) % 3).view(-1, 1)
     if half_ones:
         preds[:60] = 1
     jaccard_val = jaccard_index(
         preds=preds,
         target=target,
+        average=average,
         num_classes=3,
         ignore_index=ignore_index,
-        reduction=reduction,
+        # reduction=reduction,
```

Comment:
I am afraid, that we cannot simply remove this. I think we shall rather extend it, but not all the possibilities may always be restricted to some sort of averaging. That is why I prefer to keep this as reduction. 
Also we want this to be consistent across the entire codebase.

Even if we decide to change this to `average` which again, I am very much against, we would have to follow a proper deprecation cycle here

Reply:
@justusschock the implementation is similar to Recall and Precision implementation in the codebase, where there is no `reduction` but rather `average` which is standard for sklearn as well. The current implementation extends the functionality of using reduction - `reduction=elemntwise_mean` = `average=macro`, `reduction=None` = `average=None`, I understand depreciation-related issues, but I did not catch why we should not standardize the code like for Precision and Recall, since, let's say, they are the metrics from the same family.

Reply:
Actually, I would rather do it the other way around and have the `reduction` argument everywhere as this is more general 

cc @Borda @SkafteNicki

Reply:
well yeah found the naming issue while working on #891 as some metrics uses `average` and other `reduction`

Reply:
just raised and ask for feedback in @PyTorchLightning/core-metrics and @PyTorchLightning/core-lightning

Reply:
I agree as the average "sum" does not make much sense...

Reply:
Would this work? 
- use "reduction" everywhere where applicable, standardize it's meaning (as they did in torch)
- where "reduction" is not applicable, drop it and allow the metric to choose their own semantics, e.g., "average". For example, for precision a "reduction=sum" does not make sense.

Reply:
I would say do the update in parallel PR to all other arguments, and then come back to this PR...
cc: @justusschock for suggestions?